test/OpenTelemetry.Tests/Metrics/MetricAPITest.cs

```diff
@@ -467,6 +467,242 @@ long GetAggregatedValue(Metric metric)
+                .AddMeter(meter.Name)
+                .AddView((instrument) =>
+                {
+                    return new ExplicitBucketHistogramConfiguration { Boundaries = new[] { 5.0, 10.0 } };
+                })
+                .AddView((instrument) =>
+                {
+                    return new ExplicitBucketHistogramConfiguration { Boundaries = new[] { 10.0, 20.0 } };
+                })
+                .AddInMemoryExporter(exportedItems);
+
+            using var meterProvider = meterProviderBuilder.Build();
+
+            var instrument1 = meter.CreateHistogram<long>("name");
+            var instrument2 = meter.CreateHistogram<long>("name");
+
+            instrument1.Record(15);
+            instrument2.Record(15);
+
+            meterProvider.ForceFlush(MaxTimeToAllowForFlush);
+
+            // The second view results in a name conflict, so it is ignored
+            Assert.Single(exportedItems);
+            var metric1 = exportedItems[0];
+
+            Assert.Equal("name", exportedItems[0].Name);
+
+            var metricPoints = new List<MetricPoint>();
+            foreach (ref readonly var mp in metric1.GetMetricPoints())
+            {
+                metricPoints.Add(mp);
+            }
+
+            Assert.Single(metricPoints);
+            var metricPoint = metricPoints[0];
+
+            // Both instruments record
+            Assert.Equal(2, metricPoint.GetHistogramCount());
+            Assert.Equal(30, metricPoint.GetHistogramSum());
+
+            var index = 0;
+            var actualCount = 0;
+            var expectedBucketCounts = new long[] { 0, 0, 2 };
+            foreach (var histogramMeasurement in metricPoint.GetHistogramBuckets())
+            {
+                Assert.Equal(expectedBucketCounts[index], histogramMeasurement.BucketCount);
+                index++;
+                actualCount++;
+            }
+        }
+
+        [Fact]
+        public void DuplicateInstrumentRegistration_WithViews_TwoInstruments_OneView_ResultsInConflict()
+        {
+            var exportedItems = new List<Metric>();
+
+            using var meter = new Meter($"{Utils.GetCurrentMethodName()}");
+            var meterProviderBuilder = Sdk.CreateMeterProviderBuilder()
+                .AddMeter(meter.Name)
+                .AddView((instrument) =>
+                {
+                    if (instrument.Name == "name")
+                    {
+                        return new MetricStreamConfiguration { Name = "othername", TagKeys = new[] { "key1" } };
+                    }
+                    else
+                    {
+                        return null;
+                    }
+                })
+                .AddInMemoryExporter(exportedItems);
+
+            using var meterProvider = meterProviderBuilder.Build();
+
+            var instrument1 = meter.CreateCounter<long>("name");
+            var instrument2 = meter.CreateCounter<long>("othername");
```

Comment:
mm..this part is tricky.
The 2nd instrument, when following the View spec, should result in conflict, and hence must be ignored.
We should not merge it with the instrument1.

^ My read of spec.

Reply:
I know this isn't quite "duplicate instrument registration" since technically the instruments are not duplicates - and it is only until the view is used to create a duplicate stream, but this is the part of the spec that makes me think it was intended for every instrument registration to result in a recording instrument.

https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/api.md#instrument-type-conflict-detection

> When more than one Instrument of the same name is created for identical Meters, denoted duplicate instrument registration, the implementation MUST create a valid Instrument in every case. Here, "valid" means an instrument that is functional and can be expected to export data, despite potentially creating a [semantic error in the data model](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/datamodel.md#opentelemetry-protocol-data-model-producer-recommendations).

Reply:
https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/sdk.md#resolving-duplicate-instrument-registration-conflicts

"The implementation MUST aggregate data from identical Instruments together in its export pipeline". -- This part of the spec says merging instruments together, but this is only for identical instruments. In this case, the instruments are not identical. User made a mistake in their View config, and we should write an error log about it, and ignore instrument2.

Reply:
And since the second instrument does not match a view then I think this part of the view spec applies

> If the Instrument could not match with any of the registered View(s), the SDK SHOULD enable the instrument using the default aggregation and temporality.

Reply:
Okay I can see this can be interpreted differently. based on this, then 2nd instrument should be allowed to proceed with default view. So we should get 2 streams. The 2nd instrument should do defaults. 1st instrument should aggregate based on view config.
Ofcourse this results in duplicate stream in the exporters.